### DISCUSSION TITLE
When passing data to child, Can I skip declaring variables in mount()

### DISCUSSION THREAD

**Discussion by ouss4m4** [livewire/livewire#5530] [1. Help]
Hello

Which of the following is the convenient (recommended) way of rendering a child? 

```blade
 @livewire('dashboard.clients', ['selectedClient' => $selectedClient, 'selectedDate' => $selectedDate], 
     key(selectedClient) )
```

OR

```blade
<livewire:dashboard.clients wire:key="$selectedClient" 
:selectedClient="$selectedClient" :selectedDate="$selectedDate" />
```

and once this child is rendered, Is it okay to not have these variables in the mount function? 

```php
class ClientData
{
  public $selectedDate; 
  public $selectedClient;
  ...

  public function mount()  //<-- not decalring $selectedClient here
  { 
      $this->setDateSetup($this->selectedDate);
      $this->setSelectedCampaign($this->selectedClient);
   }

  public function render(){
  ...
  }
}
```

On similar components that don't require function calling inside mount() I don't even declare the mount function, I let livewire hook it up. I could not find something similar in the documentation or the forum

Thank you

**Comment by joshhanley** [comment]
@ouss4m4 yeah you should be able to omit the mount method in a child component. Does that not work for you?
